types/n3/index.d.ts

```diff
@@ -13,87 +13,104 @@ import * as RDF from "rdf-js";
 import { EventEmitter } from "events";
 
 export interface Prefixes {
-    [key: string]: RDF.NamedNode;
+    [key: string]: NamedNode;
 }
 
-export class Term implements RDF.Term {
-    termType: "NamedNode" | "BlankNode" | "Literal" | "Variable" | "DefaultGraph";
-    id: string;
-    value: string;
-    constructor(iri: string);
-    toJSON(): string;
-    equals(other: RDF.Term): boolean;
-    static subclass(type: any): void;
-}
+export type Term = NamedNode | BlankNode | Literal | Variable | DefaultGraph;
 
-export class NamedNode extends Term implements RDF.NamedNode {
+export class NamedNode implements RDF.NamedNode {
     termType: "NamedNode";
     value: string;
     constructor(iri: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class BlankNode extends Term implements RDF.BlankNode {
+export class BlankNode implements RDF.BlankNode {
     static nextId: number;
     termType: "BlankNode";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Variable extends Term implements RDF.Variable {
+export class Variable  implements RDF.Variable {
     termType: "Variable";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Literal extends Term implements RDF.Literal {
+export class Literal implements RDF.Literal {
     static readonly langStringDatatype: NamedNode;
     termType: "Literal";
     value: string;
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
     language: string;
-    datatype: RDF.NamedNode;
+    datatype: NamedNode;
```

Comment:
`RDF.NamedNode`

Reply:
This datatype is part of the return value of the `literal` function in the factory. The N3 factory returns a n3 NamedNode. Any reason why you'd like this on to be of type `RDF.NamedNode`?

Reply:
The reason is because `literal` in the datafactory should also take an `RDF. NamedNode` as datatype argument. This is because any kind of named node should be allowed to define the datatype, not only those that were created by N3.

Reply:
afaik the literal function would _always_ return a an N3 NamedNode (see https://github.com/rdfjs/N3.js/blob/master/lib/N3DataFactory.js#L279 , where if the datatype is a `RDF.NamedNode` it would be re-instantiated as a new N3.NamedNode).
Looking at this, I think the datatype input argument of the `literal` datafactory function should be changed from N3.NamedNode to RDF.NamedNode

Reply:
I'm assuming this is resolved? To clarify: this is the typing of the N3 Literal, which is created by the N3 datafactory. As a result, it'll always have a N3 NamedNode as datatype

Reply:
A, missed your :+1:  here. Considering it resolved ;)